samples/crypto/persistent_key_usage/src/main.c

```diff
@@ -15,20 +15,91 @@
 #include <tfm_ns_interface.h>
 #endif
 
-#define APP_SUCCESS		(0)
-#define APP_ERROR		(-1)
+#define APP_SUCCESS	    (0)
+#define APP_ERROR	    (-1)
 #define APP_SUCCESS_MESSAGE "Example finished successfully!"
-#define APP_ERROR_MESSAGE "Example exited with error!"
+#define APP_ERROR_MESSAGE   "Example exited with error!"
 
-#define PRINT_HEX(p_label, p_text, len)\
-	({\
-		LOG_INF("---- %s (len: %u): ----", p_label, len);\
-		LOG_HEXDUMP_INF(p_text, len, "Content:");\
-		LOG_INF("---- %s end  ----", p_label);\
+#define PRINT_HEX(p_label, p_text, len)                                                            \
+	({                                                                                         \
+		LOG_INF("---- %s (len: %u): ----", p_label, len);                                  \
+		LOG_HEXDUMP_INF(p_text, len, "Content:");                                          \
+		LOG_INF("---- %s end  ----", p_label);                                             \
 	})
 
 LOG_MODULE_REGISTER(persistent_key_usage, LOG_LEVEL_DBG);
 
+/* Use settings with littlefs backend */
+#ifdef CONFIG_TRUSTED_STORAGE
+
+#include "settings/settings_file.h"
+#include <zephyr/device.h>
+#include <zephyr/fs/fs.h>
+#include <zephyr/fs/littlefs.h>
+
+#define STORAGE_PARTITION    storage_partition
+#define STORAGE_PARTITION_ID FIXED_PARTITION_ID(STORAGE_PARTITION)
+
+/* LittleFS work area strcut */
+FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(cstorage);
+static struct fs_mount_t littlefs_mnt = {
+	.type = FS_LITTLEFS,
+	.fs_data = &cstorage,
+	.storage_dev = (void *)STORAGE_PARTITION_ID,
+	.mnt_point = "/0",
+};
+
+static int setup_settings_backend(void)
+{
+	int rc;
+
+	rc = fs_mount(&littlefs_mnt);
+	if (rc != 0) {
+		return rc;
+	}
+
+	return 0;
+}
+
+SYS_INIT(setup_settings_backend, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
+
+#ifdef CONFIG_TRUSTED_STORAGE_TRUSTED_AEAD_BACKEND_KEY_DERIVE_FROM_HUK
+#include <nrf_cc3xx_platform.h>
+#include <hw_unique_key.h>
+
+#ifndef HUK_HAS_KMU
+#include <zephyr/sys/reboot.h>
+#endif
+
+static int write_huk(void)
+{
+	int result = nrf_cc3xx_platform_init();
```

Comment:
nrf_cc3xx_platform_init should be handled by HW_CC3XX option, which also handles the case where entropy has not been enabled, i.e calling another init function.